src/lib/classes/access-control-list.js

```diff
@@ -0,0 +1,333 @@
+
+    quads.forEach(quad => writer.addQuad(quad));
+    let turtle;
+    writer.end((error, result) => {
+      if (!error) turtle = result;
+    });
+    return turtle;
+  };
+
+  /**
+   * @function createACLFile Creates a file or a container with it specfic set of acls. Assigns READ, WRITE and CONTROL permissions to the owner by default
+   * @param {Array<Permissions> | null} permissions Array of permissions to be added in the acl file
+   */
+  createACLFile = async (permissions = null) => {
+    await this.createSolidFile(this.documentUri);
+    if (permissions) {
+      const permissionList = [
+        { agents: this.owner, modes: [PERMISSIONS.READ, PERMISSIONS.WRITE, PERMISSIONS.CONTROL] },
+        ...permissions
+      ];
+      const body = this.createPermissionsTurtle(permissionList);
+      await this.createSolidFile(this.aclUri, { body });
+    }
+  };
+
+  /**
+   * @function void Helper function to create a file
+   * @param {String} url Url where the solid file has to be created
+   * @param {Object} options Options to add as part of the native fetch options object
+   */
+  createSolidFile = async (url: String, options: Object = {}) =>
+    solid.fetch(url, {
+      method: 'PUT',
+      headers: {
+        'Content-Type': 'text/turtle'
+      },
+      ...options
+    });
+
+  /**
+   * @function getParentACL Recursively tries to retrieve the acl file from the parent containers if there is not a direct one
+   * @param {String} url Url of the parent container
+   * @returns {Object} Parent acl fetched file
+   */
+  getParentACL = async (url: String) => {
+    const newURL = new URL(url);
+    const { pathname } = newURL;
+    const hasParent = pathname.length > 1;
+    if (!hasParent) return null;
+    const isContainer = pathname.endsWith('/');
+    let newPathname = isContainer ? pathname.slice(0, pathname.length - 1) : pathname;
+    newPathname = `${newPathname.slice(0, newPathname.lastIndexOf('/'))}/`;
+    const parentURI = `${newURL.origin}${newPathname}`;
+    const result = await solid.fetch(`${parentURI}.acl`);
+    if (result.status === 404) return this.getParentACL(parentURI);
+    if (result.status === 200) return result;
+  };
+
+  /**
+   * @function getACLFile Retrieves the acl file from the network
+   * @returns {Object} Acl fetched file
+   */
+  getACLFile = async () => {
+    try {
+      const result = await solid.fetch(this.aclUri);
+      if (result.status === 403) throw new Error(result.message || 'Not authorized');
+      if (result.status === 404) return this.getParentACL(this.documentUri);
+      return result;
+    } catch (e) {
+      throw e;
+    }
+  };
+
+  /**
+   * @function getSubjects Creates an object based on a ldflex proxy document
+   * @param {Proxy} document The base ldflex proxy document from where the data will be extracted
+   * @returns {Array<Permissions>} A custom object to visualized the acls of a document or container it a Pod
```

Comment:
Typo, should read "visualize", and I believe the end should read "or container in a Pod"?

Reply:
Done